examples/func_defs.py

```diff
@@ -28,10 +29,16 @@ def visit_FuncDef(self, node):
 
 
 def show_func_defs(filename):
-    # Note that cpp is used. Provide a path to your own cpp or
+    # Note that cpp is used for all systems but Darwin, where cpp
+    # is not usable. Provide a path to your own cpp or
     # make sure one exists in PATH.
-    ast = parse_file(filename, use_cpp=True,
-                     cpp_args=r'-Iutils/fake_libc_include')
+    if platform.system() == 'Darwin':
+        ast = parse_file(filename, use_cpp=True,
+                 cpp_path='gcc',
+                 cpp_args=['-E', '-Iutils/fake_libc_include'])
+    else:
+        ast = parse_file(filename, use_cpp=True,
+                         cpp_args=r'-Iutils/fake_libc_include')
```

Comment:
I'm not sure I follow, sorry. You enable cpp_supported for Darwin lower down, which causes a bunch of tests to execute cpp but here you specifically use gcc -E for Darwin?

Reply:
Are there `cpp` calls below on macOS? Basically `cpp` on macOS does not handle C comments, so if by chance cpp is used in other examples, they might work ok if they have no C-style comments.